src/VectorizedSolver.cpp

```diff
@@ -322,76 +328,52 @@ void VectorizedSolver::normalizeFluxes() {
 void VectorizedSolver::computeFSRSources() {
 
   int tid;
-  FP_PRECISION scatter_source;
-  FP_PRECISION fission_source;
-  FP_PRECISION* nu_sigma_f;
-  FP_PRECISION* sigma_s;
-  FP_PRECISION* sigma_t;
-  FP_PRECISION* chi;
   Material* material;
+  FP_PRECISION* sigma_t;
+  FP_PRECISION* sigma_s;
+  FP_PRECISION* fiss_mat;
+  FP_PRECISION scatter_source, fission_source;
 
-  int size = _num_FSRs * _num_groups * sizeof(FP_PRECISION);
+  int size = _num_threads * _num_groups * sizeof(FP_PRECISION);
   FP_PRECISION* fission_sources = (FP_PRECISION*)MM_MALLOC(size, VEC_ALIGNMENT);
-  size = _num_threads * _num_groups * sizeof(FP_PRECISION);
   FP_PRECISION* scatter_sources = (FP_PRECISION*)MM_MALLOC(size, VEC_ALIGNMENT);
 
   /* For all FSRs, find the source */
-  #pragma omp parallel for private(material, nu_sigma_f, chi, \
-    sigma_s, sigma_t, fission_source, scatter_source) schedule(guided)
+  #pragma omp parallel for private(tid, material, sigma_t, sigma_s, fiss_mat, \
+    fission_source, scatter_source) schedule(guided)
```

Comment:
A possibly cleaner approach might be to make thread private allocations inside the parallel region, which should reduce the chances of false sharing. It also eliminates the need to manually index each thread into the shared array, making the code a little simpler. This can be accomplished by breaking the OMP parallel pragma and OMP for pragma apart. For instance,
```
#pragma omp parallel default(none) private(...) shared(...)
{
  int size = _num_groups * sizeof(FP_PRECISION);
  FP_PRECISION * fission_sources = (FP_PRECISION*) MM_MALLOC(size, VEC_ALIGNMENT);
  #pragma omp for schedule (guided)
  for( ){
  }
}
```